docs/configuration_options.md

```diff
@@ -0,0 +1,303 @@
+# Configuration Reference
+
+TODO: SECTIONS(DELAYED JOB SUPPORT, ASYNC SUPPORT, WRITE TO FILE, PERSON TRACKING)
+
+<!-- Sub:[TOC] -->
+
+## access_token
+
+**Required**
+
+Sets the access token used to send payloads to rollbar.
+
+Items sent through a given access token arrive in that access token's project
+and respect the rate limits set on that access token.
+
+## async_handler
+
+If `config.use_async = true` explicitly sets the function used to send
+asynchronous payloads to Rollbar. Not needed if using one of the built in async
+reporters:
+
+ * girl_friday
+ * sucker_punch
+ * Sidekiq
+ * Resque
+ * threading
+
+## branch
+
+**Default** "master"
+
+Name of the checked-out source control branch.
+
+## code_version
+
+A string, up to 40 characters, describing the version of the application code.
+
+Rollbar understands these formats:
+
+ * semantic version (i.e. "2.1.12")
+ * integer (i.e. "45")
+ * git SHA (i.e. "3da541559918a808c2402bba5012f6c60b27661c")
+
+## custom_data_method
+
+TODO: What does it do? Demo?
+
+## delayed_job_enabled
+
+**Default** `true`
+
+TODO: Verify
+
+Set to false if you have `Delayed`  but do not wish to wrap Delayed jobs with a
+Rollbar notifier.
+
+## default_logger
+
+**Default** `Logger.new(STDERR)`
+
+What logger to use for printing debugging and informational messages during
+operation.
+
+## disable_monkey_patch
+
+**Default** `false`
+
+Disables monkey patching `Rack::Builder`. If you do this you'll need to manually
+`use` an appropriate Rollbar middleware.
+
+Especially useful if you're not using Rails or Sinatra.
+
+## disable_core_monkey_patch
+
+**Default** `false`
+
+TODO: VERIFY
+
+Disables monkey patching certain core features without which Rollbar may not
```

Comment:
For now this disables the basic socket monkey patch if active support is defined. Rollbar can work without it except in some specific cases.

There is another monkey patch, to Thread, that cannot be removed and it's always applied.